components/core/src/ffi/ir_stream/decoding_methods.cpp

```diff
@@ -329,8 +347,7 @@ namespace ffi::ir_stream {
                 {
                     return error_code;
                 }
-                all_dict_var_strings.append(var_str);
-                dictionary_var_end_offsets.push_back(all_dict_var_strings.length());
+                dict_vars.push_back(var_str);
```

Comment:
Minor:
* There's a chance that `emplace_back` may have better performance.
* You might also consider moving `var_str` out of the loop.